### PR TITLE
Npgsql 6.0 release and breaking change notes

### PR DESCRIPTION
**Pull request by roji**
Note that this doesn't update [the date/time mappings page](https://www.npgsql.org/doc/types/datetime.html). We should absolutely do that, but the release/breaking changes notes seem sufficient for now in order to release rc1.

I also plan to do a friendly blog post detailing all the timestamp mess.

### CODE REVIEW

File: conceptual/Npgsql/release-notes/6.0.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+* When reading `timestamptz` as [ZonedDateTime](https://nodatime.org/3.0.x/api/NodaTime.ZonedDateTime.html) or [OffsetDateTime](https://nodatime.org/3.0.x/api/NodaTime.OffsetDateTime.html), UTC values are always returned. Previously, local values based on the PostgreSQL `TimeZone` parameter were returned.
+
+> [!NOTE]
+> In most cases, storing UTC timestamps is the recommended practice. If this is what you're doing, it's strongly recommended to migrate all relevant columns from `timestamp` columns to `timestamptz`. See below for how to do this.
+
+#### Opting out of the new timestamp mapping logic
+
+The changes described above are far-reaching, and may break applications in various ways. You can upgrade to version 6.0 but opt out of the new mapping by enabling the `Npgsql.EnableLegacyTimestampBehavior` [AppContext switch](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext?view=net-5.0):
+
+To revert to the legacy timestamp behavior, add the following at the start of your application, before any EF Core operations are invoked:
```
Comment (NinoFloris) [inline conceptual/Npgsql/release-notes/6.0.md:107]:
```suggestion
To revert to the legacy timestamp behavior, add the following at the start of your application, before any Npgsql operations are invoked:
```

File: conceptual/Npgsql/release-notes/6.0.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+* It is no longer possible to write [DateTimeOffset](https://docs.microsoft.com/dotnet/api/system.datetimeoffset) with offsets other than 0 (UTC), since these cannot be represented in PostgreSQL. These were previously implicitly converted to UTC before sending.
+* It is no longer possible to read or write `timetz` as DateTime or TimeSpan, as these don't have a timezone. This was previously allowed, with the offset being stripped.
+
+See the [EF Core provider docs](/efcore/release-notes/6.0.html) for additional changes at the EF level.
+
+#### NodaTime changes
+
+* [Instant](https://nodatime.org/3.0.x/api/NodaTime.Instant.html) is now sent as a `timestamptz` value, and not as a `timestamp`, since they represent a universally agreed-upon point in time. To send a `timestamp`, use [LocalDateTime](https://nodatime.org/3.0.x/api/NodaTime.LocalDateTime.html).
+* When reading `timestamptz` as [ZonedDateTime](https://nodatime.org/3.0.x/api/NodaTime.ZonedDateTime.html) or [OffsetDateTime](https://nodatime.org/3.0.x/api/NodaTime.OffsetDateTime.html), UTC values are always returned. Previously, local values based on the PostgreSQL `TimeZone` parameter were returned.
```
Comment (NinoFloris) [inline conceptual/Npgsql/release-notes/6.0.md:98]: Pg timezone or machine timezone? Was there really a difference between nodatime and bcl handling?

**Review by NinoFloris — APPROVED** [review]

File: conceptual/Npgsql/release-notes/6.0.md (hunk 2 above)
Comment (roji) [reply conceptual/Npgsql/release-notes/6.0.md:98]: Yeah, there actually was... The BCL types could only work with Windows timezones (until .NET 6.0 that is), whereas NodaTime has always supported IANA/Olson, so we actually looked at TimeZone in an effort to be even more conforming to the PG behavior... All seems pretty senseless now.

Comment (NinoFloris) [reply conceptual/Npgsql/release-notes/6.0.md:98]: TIL :o

Comment (roji) [reply conceptual/Npgsql/release-notes/6.0.md:98]: We even had https://github.com/npgsql/npgsql/issues/3594 :rofl:

**Pull request merged**